### ISSUE TITLE
Fuzzy version matching not working as expected with golang@1.19

### ISSUE THREAD

**Issue by amoosbr** [jdxcode/rtx#120]
Using `golang 1.19` in `.tool-versions` will not install and use the actual latest golang 1.19 release.

Actual:
```sh
❯ rtx global --fuzzy golang@1.19
❯ rtx exec golang@1.19 -- go version
go version go1.19 darwin/arm64
```

Expected:
```sh
❯ rtx global --fuzzy golang@1.19
❯ rtx exec golang@1.19 -- go version
go version go1.19.5 darwin/arm64
```

This is probably related to the fix of not doing a fuzzy search, when there is an actual release matching the release number and golang not using a strict x.y.z version numbering scheme.
And it might not be something, we're able to fix without changes to `.tool-versions` syntax.

This issue might therefore be just for documentation purposes.

Work-around (add a dot to the end of the version) :-):
```sh
❯ rtx exec golang@1.19. -- go version
go version go1.19.5 darwin/arm64
```

**Comment by jdxcode** [comment]
ohh I see. This one is more complicated because golang doesn't use `1.19.0` so it breaks our prefix matching. I'm not sure what we need to do for this one yet.

**Comment by jdxcode** [comment]
I think you're right, we may need to do something in `.tool-versions` like this:

```
golang prefix:1.19
```

**Comment by jdxcode** [comment]
the workaround isn't a proper solution since it would only work if at least 1 patch version exists for a runtime. I'm sure you already know this but I thought it was good to point out anyways.

**Comment by jdxcode** [comment]
Maybe one solution would be to somehow dispatch pattern matching to a plugin

**Comment by jdxcode** [comment]
seems we're further along than asdf here (unless I'm doing something wrong):

```
~ ❯ asdf latest golang 1.2
No compatible versions available (golang 1.2)
~ ❯ asdf latest golang 1.1
No compatible versions available (golang 1.1)
~ ❯ asdf latest golang 1.19
No compatible versions available (golang 1.19)
~ ❯ asdf latest golang
No compatible versions available (golang [0-9])
```

**Comment by amoosbr** [comment]
Yeah, I already thought this might not be easy or worth to work on. As you guessed, I know that I need to wait for the first patch version, for my workaround to work. But I think, this is fine.
Thanks for checking, how asdf was handling this.

In my opinion, this issue can be closed. If another user searches for this, he can find this discussion and workaround. And if he has a valid reason, can either reopen or create a new issue.

**Comment by jdxcode** [comment]
I think it deserves to stay open, this is a definite issue. Perhaps someone will come along with a smart idea on resolving it.

**Comment by roele** [comment]
Running into this while trying to uninstall the old version `1.20` which removes the newly installed patch version `1.20.1`

**Comment by jdxcode** [comment]
that's not surprising. The behavior of `rtx uninstall` I've never really decided on when it comes to prefixes. Perhaps for now I should take out the prefix matching completely so you'd always have to specify exact versions for uninstall.

**Comment by jdxcode** [comment]
if it's screwing with your workflow for any reason just do `rm -rf ~/.local/share/rtx/installs/golang/1.20`. I am working on a big refactor that will touch this so I should be able to get around to fixing it soon.

**Comment by jdxcode** [comment]
#145 will fix the uninstall issue

**Comment by jdxcode** [comment]
the whole golang thing is also fixed in #145 but I haven't documented it yet and it isn't working everywhere. It also may change a bit. If you set `golang prefix:1.20`, then it will behave like you want it to.
